### PR TITLE
feat(console): init organization settings

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

init console organization settings page

image

## Testing

manually tested, will add integration tests later

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/ds-components/Tag/index.module.scss [hunk 1]
```diff
@@ -21,6 +21,16 @@
     color: var(--color-on-success-container);
   }
 
+  // Distinguish from the info status
+  &.cell {
+    font-family: 'Roboto Mono', monospace;
+    font-size: 14px;
+    line-height: 20px;
+    border-radius: 6px;
+    background: var(--color-neutral-variant-90);
```
Comment (charIeszhao) [inline packages/console/src/ds-components/Tag/index.module.scss:30]:
```suggestion
    background: var(--color-bg-info-tag);
```

File: packages/console/src/ds-components/Tag/index.module.scss [hunk 2]
```diff
@@ -21,6 +21,16 @@
     color: var(--color-on-success-container);
   }
 
+  // Distinguish from the info status
+  &.cell {
+    font-family: 'Roboto Mono', monospace;
+    font-size: 14px;
+    line-height: 20px;
```
Comment (charIeszhao) [inline packages/console/src/ds-components/Tag/index.module.scss:28]:
```suggestion
    font: var(--font-body-2);
    font-family: 'Roboto Mono', monospace;
```

File: packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  } = useForm<{ name: string; description?: string }>({ defaultValues: { name: '' } });
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const addPermission = handleSubmit(async (json) => {
+    setIsLoading(true);
+    try {
+      await api.post('api/organization-scopes', {
+        json,
+      });
+      onFinish();
```
Comment (charIeszhao) [inline packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx:35]: nit: consider returning the response json to `onFinish()` callback, in order to reduce a `GET` request on `mutate`

File: packages/console/src/pages/Organizations/PermissionsField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            dataIndex: 'description',
+            colSpan: 6,
+            render: ({ description }) => description ?? '-',
+          },
+          {
+            title: null,
+            dataIndex: 'delete',
+            render: ({ id }) => (
+              <DeleteButton
+                content="Delete at your own risk, mate."
```
Comment (charIeszhao) [inline packages/console/src/pages/Organizations/PermissionsField/index.tsx:75]: 😂

File: packages/console/src/components/DeleteButton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      await onDelete();
+    } finally {
+      setIsDeleting(false);
+      setIsModalOpen(false);
+    }
+  }, [onDelete]);
+
+  return (
+    <>
+      <Tooltip content={<div>{t('general.delete')}</div>}>
```
Comment (charIeszhao) [inline packages/console/src/components/DeleteButton/index.tsx:37]: I've seen other usages in console and the tooltip shows "Remove". Should double check and unify these use cases

File: packages/console/src/ds-components/Select/index.module.scss [hunk 6]
```diff
@@ -16,6 +16,40 @@
...
+      cursor: auto;
+      display: flex;
+      align-items: center;
+      gap: _.unit(1);
+    }
+
+    .close {
+      width: 16px;
+      height: 16px;
+    }
```
Comment (charIeszhao) [inline packages/console/src/ds-components/Select/index.module.scss:36]: `cursor: pointer?`
Or maybe we can make the entire `.tag` show the pointer cursor on mouse hover

File: packages/console/src/ds-components/Table/index.tsx [hunk 7]
```diff
@@ -102,6 +113,7 @@ function Table<
             isEmpty && styles.empty,
             bodyTableWrapperClassName
           )}
+          style={{ padding: `0 ${padding}px ${padding}px` }}
```
Comment (charIeszhao) [inline packages/console/src/ds-components/Table/index.tsx:116]: Any reason we add this custom padding to the table? A design change?

File: packages/console/src/pages/Organizations/PermissionsField/index.tsx (hunk 4 above)
Comment (charIeszhao) [reply packages/console/src/pages/Organizations/PermissionsField/index.tsx:75]: bruh

File: packages/console/src/pages/Organizations/index.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+const pathnames = Object.freeze({
+  organizations: 'organizations',
+  settings: 'settings',
+});
+
+function Organizations() {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { tab } = useParams();
+
+  console.log('tab', tab);
```
Comment (charIeszhao) [inline packages/console/src/pages/Organizations/index.tsx:21]: nit: should remove

File: packages/console/src/scss/_underscore.scss [hunk 9]
```diff
@@ -96,3 +96,14 @@
...
 }
+
+/** Render a tag that has background color and border radius. */
+@mixin tag {
+  display: inline-block;
+  max-width: 100%;
+  vertical-align: bottom;
+  padding: unit(1) unit(2);
+  border-radius: 6px;
+  background: var(--color-neutral-95);
```
Comment (charIeszhao) [inline packages/console/src/scss/_underscore.scss:107]: Should we use `--color-bg-info-tag`?

**Review by charIeszhao — APPROVED** [review]
Overall looks OK. Approved with comments

File: packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx (hunk 3 above)
Comment (gao-sun) [reply packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx:35]: we don't have the convention to save the `GET` request currently. shall we proceed?

File: packages/console/src/pages/Organizations/PermissionsField/index.tsx (hunk 4 above)
Comment (gao-sun) [reply packages/console/src/pages/Organizations/PermissionsField/index.tsx:75]: will update soon

File: packages/console/src/components/DeleteButton/index.tsx (hunk 5 above)
Comment (gao-sun) [reply packages/console/src/components/DeleteButton/index.tsx:37]: let's talk to our designers

File: packages/console/src/ds-components/Select/index.module.scss (hunk 6 above)
Comment (gao-sun) [reply packages/console/src/ds-components/Select/index.module.scss:36]: it's a button. already pointer

Comment (gao-sun) [reply packages/console/src/ds-components/Select/index.module.scss:36]: nah, tag is tag

File: packages/console/src/pages/Organizations/index.tsx (hunk 8 above)
Comment (gao-sun) [reply packages/console/src/pages/Organizations/index.tsx:21]: removed later

File: packages/console/src/scss/_underscore.scss (hunk 9 above)
Comment (gao-sun) [reply packages/console/src/scss/_underscore.scss:107]: this mixin is removed in the upcoming PRs

File: packages/console/src/components/DeleteButton/index.tsx (hunk 5 above)
Comment (gao-sun) [reply packages/console/src/components/DeleteButton/index.tsx:37]: started Slack thread

File: packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx (hunk 3 above)
Comment (charIeszhao) [reply packages/console/src/pages/Organizations/CreatePermissionModal/index.tsx:35]: OK for me

File: packages/console/src/ds-components/Table/index.tsx (hunk 7 above)
Comment (charIeszhao) [reply packages/console/src/ds-components/Table/index.tsx:116]: Try `hasBorder` prop on Table

**Review by simeng-li — APPROVED** [review]
LGTM on core updates

**Pull request merged**